driver-core/src/main/java/com/datastax/driver/core/EventDebouncer.java

```diff
@@ -151,9 +155,9 @@ private void completeAllPendingFutures() {
 
         if (state == State.RUNNING) {
             int count = eventCount.get();
-            if (count < maxPendingEvents) {
+            if (count < maxPendingEvents()) {
                 scheduleDelayedDelivery();
-            } else if (count == maxPendingEvents) {
+            } else if (count == maxPendingEvents()) {
                 scheduleImmediateDelivery();
```

Comment:
If `maxPendingEvents()` is decreased dynamically, we might miss that condition and not schedule an immediate delivery (since now `count > maxPendingEvents()`).
Not sure if that's a huge issue since a delayed delivery should still be scheduled.